django/contrib/contenttypes/fields.py

```diff
@@ -226,8 +224,8 @@ def __get__(self, instance, cls=None):
         pk_val = getattr(instance, self.fk_field)
 
         try:
-            rel_obj = getattr(instance, self.cache_attr)
-        except AttributeError:
+            rel_obj = self.get_cached_value(instance)
```

Comment:
While I agree it makes the API more friendlier, internally None is handled differently than a missing key.
Meaning that None is a valid cache value where as missing key always indicates that the object does not have that field cached.

In the code block above, this would fit in nicely but in most of the other places, there has to be a clear distinction between a missing key and a value of None.